baseplate/server/healthcheck.py

```diff
@@ -13,25 +15,30 @@
 from baseplate.lib.config import InternetAddress
 from baseplate.lib.thrift_pool import ThriftConnectionPool
 from baseplate.thrift import BaseplateService
+from baseplate.thrift.ttypes import IsHealthyProbe
+from baseplate.thrift.ttypes import IsHealthyRequest
 
 
 TIMEOUT = 30  # seconds
 
 
-def check_thrift_service(endpoint: EndpointConfiguration) -> None:
+def check_thrift_service(endpoint: EndpointConfiguration, probe: int) -> None:
     pool = ThriftConnectionPool(endpoint, size=1, timeout=TIMEOUT)
     with pool.connection() as protocol:
         client = BaseplateService.Client(protocol)
-        assert client.is_healthy(), "service indicated unhealthiness"
+        assert client.is_healthy(
+            request=IsHealthyRequest(probe=probe),
+        ), "service indicated unhealthiness in probe {probe}"
 
 
-def check_http_service(endpoint: EndpointConfiguration) -> None:
+def check_http_service(endpoint: EndpointConfiguration, probe: int) -> None:
+    probe_str = IsHealthyProbe._VALUES_TO_NAMES[probe].lower()
```

Comment:
1. the design doc used string instead of int
2. technically changing the string name of an enum is a breaking change, so I don't agree that int values have stronger guarantee than string values to not change
3. string values are really more human readable.

Reply:
> 1. the design doc used string instead of int

I don't think that matters?

> 2. technically changing the string name of an enum is a breaking change, so I don't agree that int values have stronger guarantee than string values to not change

It's a breaking change when _you_ update your IDL but if baseplate-healthcheck starts coming from something that is separate from your service (or just has the thrift file updated separately) the enum name changing in there shouldn't break your service.

> 3. string values are really more human readable.

I don't think that's really relevant here? Humans aren't sending these requests.  We could add a `ProbeFromRequest` helper to get the value and convert it from a string to an int?

Reply:
i slightly lean towards strings for human friendliness. maybe add a comment to the IDL saying the names can't be changed safely?

Reply:
Having been bit by this sort of thing _several_ times in the past I would encourage that you consider using the ints rather than strings, I don't think that human readability is as important in something like this and means you don't even need that comment (which can easily be ignored).

Reply:
Using ints will cause a burden for services implementing http health check. They kind of need to double check again and again to make sure they didn't get the number to meaning mapping wrong. String names, on the other hand, comes from well defined and well documented kubernetes, so it's both less likely to get wrong (I can see people misspell "readiness", though), and less likely to change at will.

Who changes enum names at will anyways? That's asking for troubles.

Reply:
> Using ints will cause a burden for services implementing http health check.

I don't think it really does?

```python
def is_healthy(request):
    probe = baseplate.GetHTTPProbe(request)
    if probe is IsHealthyProbe.READINESS:
    ...
```

Seems better to me than having to check:

```
if probe == "readiness"
```

in every service.

Reply:
Note, you can still use the strings in the CLI argument, I'm just advocating for using the int values in the actual request.

Reply:
that assumes the http service actually have the compiled thrift file, which I don't think is true for most cases?

but if that's actually true for most http service code, then I agree ints are better.

Reply:
for python at least we do have the compiled thrift always available. that code example is pretty compelling. but i would be sad to lose the readability of the http request for operators' log perusing purposes.

Reply:
ok. changed to int.